lib/utils/device/live.ts

```diff
@@ -105,12 +106,14 @@ export class LivepushManager {
 			// We only care about builds
 			if (service.build != null) {
 				const context = path.join(this.buildContext, service.build.context);
-				const dockerfile = buildTask.dockerfile;
-				if (dockerfile == null) {
+				if (buildTask.dockerfile == null) {
 					throw new Error(
 						`Could not detect dockerfile for service: ${serviceName}`,
 					);
 				}
+				const dockerfile = new Dockerfile(
+					new Dockerfile(buildTask.dockerfile).generateLiveDockerfile(),
+				);
```

Comment:
Not sure if there is a more elegant way of doing this?

Reply:
```suggestion
const dockerfile = new Dockerfile(buildTask.dockerfile)
```

The generation happens automatically by default, and the internal representation is replaced with the live one. 
`generateLiveDockerfile` is a bit of a misnomer tbh and you only really need it when you want the string representation of the dockerfile